Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,217 +14,298 @@
+ @Status Interoperable
 */
-+ (instancetype)orderedSetWithObjects:(id)firstObj {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (instancetype)orderedSetWithObjects:(id)firstObj, ... {
+    va_list argList;
+    va_start(argList, firstObj);
+    std::vector<id> flatArgs = ConvertVAListToVector((id)firstObj, argList);
+    va_end(argList);
+
+    if (flatArgs.size() == 0) {
+        // return empty set if args are nil.
+        return [[self new] autorelease];
+    }
+
+    return [[[self alloc] initWithObjects:flatArgs.data() count:flatArgs.size()] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-+ (instancetype)orderedSetWithObjects:(id _Nonnull const[])objects count:(NSUInteger)cnt {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (instancetype)orderedSetWithObjects:(id _Nonnull const[])objects count:(NSUInteger)count {
+    return [[[self alloc] initWithObjects:objects count:count] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithOrderedSet:(NSOrderedSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self orderedSetWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithOrderedSet:(NSOrderedSet*)set range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[self alloc] initWithOrderedSet:set range:range copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[self alloc] initWithSet:set] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[self alloc] initWithSet:set copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (array == nil) {
+        return [self init];
+    }
+
+    if (NSMaxRange(range) > [array count]) {
+        [NSException raise:NSInvalidArgumentException
+                    format:@"The given range length is greater than the array length: %u.", [array count]];
+    }
```

Comment:
We have some macros for this that we'd prefer that you use. THROW_NS_IF_FALSE and friends are useful for these sorts of arg checks. Note that E_INVALIDARG maps to NSInvalidArgumentException.